src/Hydrator.php

```diff
@@ -0,0 +1,245 @@
+                        $this
+                    );
+                } catch (SkipTypecastException) {
+                }
+            }
+        }
+
+        return [$excludeParameterNames, $constructorArguments];
+    }
+
+    /**
+     * @psalm-param MapType $map
+     */
+    private function getHydrateData(
+        object $object,
+        array $sourceData,
+        array $map,
+        bool $strict,
+        array $excludeProperties = []
+    ): array {
+        $hydrateData = [];
+
+        $data = $this->createData($object, $sourceData, $map, $strict);
+
+        foreach ($this->getObjectProperties($object) as $property) {
+            if (!empty($property->getAttributes(SkipHydrate::class))) {
+                continue;
+            }
+
+            $propertyName = $property->getName();
+            if (in_array($propertyName, $excludeProperties, true)) {
+                continue;
+            }
+
+            $resolved = false;
+            $resolvedValue = null;
+            try {
+                $resolvedValue = $this->resolve($propertyName, $data);
+                $resolved = true;
+            } catch (NotResolvedException) {
+            }
+
+            try {
+                $resolvedValue = $this->parameterAttributesHandler->handle(
+                    $property,
+                    $resolved,
+                    $resolved ? $resolvedValue : null,
+                    $data,
+                );
+                $resolved = true;
+            } catch (NotResolvedException) {
+            }
+
+            if ($resolved) {
+                try {
+                    $hydrateData[$propertyName] = $this->typecast->cast($resolvedValue, $property->getType(), $this);
+                } catch (SkipTypecastException) {
+                }
+            }
+        }
+
+        return $hydrateData;
+    }
+
+    /**
+     * @throws NotResolvedException
+     */
+    private function resolve(string $name, Data $data): mixed
+    {
+        $map = $data->getMap();
+
+        if ($data->isStrict() && !array_key_exists($name, $map)) {
+            throw new NotResolvedException();
+        }
+
+        return DataHelper::getValueByPath($data->getData(), $map[$name] ?? $name);
+    }
+
+    private function populate(object $object, array $values): void
+    {
+        /** @var Closure $setter */
+        $setter = Closure::bind(
+            static function (object $object, string $propertyName, mixed $value): void {
+                $object->$propertyName = $value;
+            },
+            null,
+            $object
+        );
+
+        foreach ($values as $propertyName => $value) {
+            $setter($object, $propertyName, $value);
+        }
+    }
+
+    /**
+     * @psalm-return array<string, ReflectionProperty>
+     */
+    private function getObjectProperties(object $object): array
+    {
+        $result = [];
+
+        $properties = (new ReflectionClass($object))->getProperties();
```

Comment:
Use filter to filter static properties
https://www.php.net/manual/en/class.reflectionproperty.php#reflectionproperty.constants.modifiers

Reply:
How get all non-static properties? Seems, filter do not allow it.

Reply:
Hm, seems like `~ReflectionProperty::IS_STATIC` doesn't work.